src/Workspaces/Core/Portable/Storage/AbstractPersistentStorageService.cs

```diff
@@ -40,45 +41,52 @@ protected AbstractPersistentStorageService(IPersistentStorageLocationService loc
         /// to delete the database and retry opening one more time.  If that fails again, the <see
         /// cref="NoOpPersistentStorage"/> instance will be used.
         /// </summary>
-        protected abstract IChecksummedPersistentStorage? TryOpenDatabase(Solution solution, string workingFolderPath, string databaseFilePath);
+        protected abstract IChecksummedPersistentStorage? TryOpenDatabase(SolutionKey solutionKey, Solution? solutionOpt, string workingFolderPath, string databaseFilePath);
```

Comment:
>solutionOpt [](start = 109, length = 11)

`opt` suffix is not necessary since the null annotation already indicates the parameter is optional.

Reply:
true.  i can remove in followup.